management/user.go

```diff
@@ -28,57 +27,56 @@ func UserFromTelegram(user *telegram.User, db* database.MySQL) User {
 	return u
 }
 
-
 func (u *User) GetFromDataBase(db *database.MySQL) {
-    if db.Conn == nil {
-        panic("Database connection is nil")
-    }
-    query := "SELECT * FROM users WHERE id = ?"
-    rows, err := db.Conn.Query(query, u.ID)
-    if err != nil {
-        panic(err)
-    }
-    defer rows.Close()
-    if !rows.Next() {
-        panic("User not found in database")
-    }
-    var createdAt []byte
-    var geminiAPIKey sql.NullString
-    err = rows.Scan(
-        &u.ID,
-        &u.Username,
-        &u.FName,
-        &u.LName,
-        &u.LanguageCode,
-        &createdAt,
-        &u.IsAdmin,
-        &u.SQLUsername,
-        &u.SQLPassword,
-        &u.SQLDBName,
-        &geminiAPIKey,
-    )
-    if err != nil {
-        panic(err)
-    }
-    u.CreatedAt, err = time.Parse("2006-01-02 15:04:05", string(createdAt))
-    if err != nil {
-        panic(err)
-    }
-    if geminiAPIKey.Valid {
-        u.GeminiAPIKey = geminiAPIKey.String
-    } else {
-        u.GeminiAPIKey = ""
-    }
+	if db.Conn == nil {
+		panic("Database connection is nil")
+	}
+	query := "SELECT * FROM users WHERE id = ?"
+	rows, err := db.Conn.Query(query, u.ID)
+	if err != nil {
+		panic(err)
+	}
+	defer rows.Close()
+	if !rows.Next() {
+		panic("User not found in database")
+	}
+	var createdAt []byte
+	var geminiAPIKey sql.NullString
+	err = rows.Scan(
+		&u.ID,
+		&u.Username,
+		&u.FName,
+		&u.LName,
+		&u.LanguageCode,
+		&createdAt,
+		&u.IsAdmin,
+		&u.SQLUsername,
+		&u.SQLPassword,
+		&u.SQLDBName,
+		&geminiAPIKey,
+	)
+	if err != nil {
+		panic(err)
+	}
+	u.CreatedAt, err = time.Parse("2006-01-02 15:04:05", string(createdAt))
+	if err != nil {
+		panic(err)
+	}
+	if geminiAPIKey.Valid {
+		u.GeminiAPIKey = geminiAPIKey.String
+	} else {
+		u.GeminiAPIKey = ""
+	}
 }
 
 func (u *User) AddToDataBase(db *database.MySQL) {
-    if db.Conn == nil {
-        panic("Database connection is nil")
-    }
+	if db.Conn == nil {
+		panic("Database connection is nil")
+	}
 	// Convert the ID->string->byte->hash
 	password, err := bcrypt.GenerateFromPassword([]byte(strconv.FormatInt(u.ID, 10)), bcrypt.DefaultCost)
 	if err != nil {
-	    panic(err)
+		panic(err)
 	}
 	u.SQLPassword = string(password)[:20] // Limit the password to 20 characters
```

Comment:
> **🚨 issue (security):** Do not truncate bcrypt hash as it compromises security
> 
> Truncating a bcrypt hash to 20 characters severely weakens its cryptographic properties and could lead to hash collisions. Store the full hash value.

This is not an issue because the bcrypt hash itself is used as a password, and this is limiting of characters is to done wantedly. Security is not really an issue here because the SQL database has NO root access aside from localhost. The 20 char limit exists to prevent longg passwords